x-pack/plugins/alerting/server/rules_client/lib/increment_revision.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { SavedObject } from '@kbn/core/server';
+import { get, isEqual } from 'lodash';
+import { RawRule, RuleTypeParams } from '../../types';
+import { fieldsToExcludeFromRevisionUpdates, UpdateOptions } from '..';
+
+export function incrementRevision<Params extends RuleTypeParams>(
+  currentRule: SavedObject<RawRule>,
+  { data }: UpdateOptions<Params>,
+  updatedParams: RuleTypeParams
+): number {
+  // Diff root level fields
+  for (const [field, value] of Object.entries(data).filter(([key]) => key !== 'params')) {
+    if (!fieldsToExcludeFromRevisionUpdates.map(toString).includes(field)) {
```

Comment:
`fieldsToExcludeFromRevisionUpdates` wasn't being seen as an `string[]` since originally declared as `Array<keyof Rule>`. No issue using `Set<keyof RuleTypeParams>` so this was removed in [cbd1bed](https://github.com/elastic/kibana/pull/147398/commits/cbd1bed630f3ecac3ec4c84d5e896aaf3d6a2425).